### DISCUSSION TITLE
Oscillation detection: Time resolution for FOOOFGroup outputs

### DISCUSSION THREAD

**Discussion by ErnestJohnston** [lucwilson/SPRiNT#4] [Q&A] [answered]
Hello,

I am trying out your Python implementation of SPRiNT using the `SPRiNT_py.py` script for oscillation detection, and I had a few questions:

1. At first glance, it seems like SPRiNT is an alternative method for oscillation detection similar to fBOSC, but it seems like SPRiNT does not use a frequency-specific time threshold to decide of oscillations. Am I correct in that conclusion, and how do you conceptualize the role of SPRiNT in oscillation detection as compared to another method like fBOSC (i.e., is it a direct alternative or does it fulfill a different role)?

2. I want to plot an overlay mask of the detected aperiodic activities (oscillations) onto a time-frequency spectogram. To produce such an mask, I tried to use the `fgs` array generated by SPRiNT_remove_outliers(), which contains one outlier-removed FOOOFGroup object for each channel. Unfortunately, these FOOOFGroup objects contain no information about the start and stop time of the oscillation that was detected. Where can I find this information?

3. I am unsure about how to combine SPRiNT results across trials (for each channel separately). My current plan is to run SPRiNT for each trial separately and then generate a time-frequency overlay mask (as above) that can be additively combined across trials to form a heatmap representing the sum of oscillations detected across all trials. So a value of 10 at a single time-frequency pixel would indicate that an oscillation was present on that pixel for 10 trials. Is there an alternative way combine SPRiNT results across trials?

Thank you!

**Comment by lucwilson** [comment, marked as answer]
Hi there,

Thank you for reaching out with your questions! The python version of SPRiNT is relatively new but retains all of the core elements of SPRiNT in Matlab. Here are my thoughts on each of your questions.

1. Indeed, SPRiNT possesses many similar qualities to algorithms like fBOSC and PAPTO, but the typical timescales used for SPRiNT are a little longer (usually 0.5s resolution). You may still be able to go lower, but we haven’t explicitly validated resolutions comparable to those used by burst detection algorithms. The main nuance SPRiNT provides over these methods (in my opinion) is that it samples for dynamics in aperiodic activity, which could complicate burst detection where other methods assume static aperiodic features. SPRiNT’s post-processing is conceptually similar to thresholding for bursts, but you are right in that it is not tuned for specific frequency thresholds and thus may be more/less sensitive at certain frequencies, depending on your settings.

2. SPRiNT core functionality does not organize detected spectral peaks to generate putative bursts, although there is a small function in the Matlab version that does something like this (see `cluster_peaks_dynamic2`). Generally, I’ve found it's easiest to set a frequency range to extract peaks based on their centre frequencies, and then look for continuity in their detected times (if two peaks are within the frequency range simultaneously, keep the one with highest amplitude). Combined with post-processing, this should give you approximately what you're looking for (with considerable separation between bouts of oscillatory activity, you can liken these to bursts).

3. What you describe here is certainly a reasonable way to plot the probability of observing a given observation across time. I would just be relatively liberal on your pixel frequency range (say 3-4 Hz) to ensure you’re not splitting the data vertically in two. If you want to plot the raw peak power across trials, you could always average across the `peak_fit`s (which can be generated from feeding through the recovered parameters to rebuild the spectrum fit by gaussians, if it’s not already saved in the data).

I hope these answers have helped to clarify a few of your queries, but feel free to reach out again if anything remains unclear. Best of luck with your analyses!

Luc

**Reply by ErnestJohnston** [reply]
Thank you so much! Your reply has been quite helpful.
